### PR TITLE
Added support for Oxford ITC503

### PR DESCRIPTION
**Pull request by CasperSchippers**
**Finished (I think at least) and ready for review / merge**

Added support for the Oxford Instruments Intelligent Temperature Controller (ITC) 503.

### CODE REVIEW

File: pymeasure/instruments/oxfordinstruments/itc503.py [hunk 1]
```diff
@@ -0,0 +1,305 @@
...
+
+from pymeasure.instruments import Instrument
+from pymeasure.instruments.validators import strict_discrete_set, \
+    truncated_range, strict_range
+
+
+class ITC503(Instrument):
+    """Represents the Oxford Intelligent Temperature Controller 503.
+
+    .. code-block:: python
```
Comment (cjermain) [inline pymeasure/instruments/oxfordinstruments/itc503.py]: Nice! Your documentation is awesome.

File: pymeasure/instruments/oxfordinstruments/itc503.py [hunk 2]
```diff
@@ -0,0 +1,305 @@
...
+                      is considered at set-point
+        :param timeout: The maximum time the waiting is allowed to take. If
+                        timeout is exceeded, a TimeoutError is raised. If
+                        timeout is set to zero, no timeout will be used.
+        :param check_interval: The time between temperature queries to the ITC.
+        :param stability_interval: The time over which the temperature_error is
+                                   to be below error to be considered stable.
+        :param thermalize_interval: The time to wait after stabilizing for the
+                                    system to thermalize.
+        :param should_stop: Optional function (returning a bool) to allow the
```
Comment (cjermain) [inline pymeasure/instruments/oxfordinstruments/itc503.py]: I'm glad to see the `should_stop` pattern repeated.

**Comment by CasperSchippers** [comment]
Thanks for the quick review. I added my name to the author lists

**Comment by cjermain** [comment]
@CasperSchippers, nice work! Thanks.

**Pull request merged**